data-prepper-plugins/geoip-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/MaxMindGeoIpProviderTest.java

```diff
@@ -0,0 +1,72 @@
+package com.amazon.dataprepper.plugins.prepper.geoip.provider;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.plugins.prepper.geoip.GeoIPPrepperConfig;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.internal.matchers.apachecommons.ReflectionEquals;
+
+import java.util.HashMap;
+
+
+public class MaxMindGeoIpProviderTest {
+    static PluginSetting pluginSetting;
+    private static final String PROVIDER_TYPE = "MaxMindGeolite2CityDatabase";
+    private static final String TEST_DATABASE_PATH = "./src/test/resources/GeoLite2-City-Test.mmdb";
+    private static final String TEST_BAD_DATABASE_PATH = "./src/test/resources/Corrupted-City-Test.mmdb";
+    private static final String TEST_WRONG_DATABASE_PATH = "./does/not/exist/GeoLite2-City-Test.mmdb";
+
+    public static MaxMindGeoIpProvider maxMindGeoIpProvider;
+
+    @BeforeAll
+    public static void setup() {
+        pluginSetting = new PluginSetting(
+                "geoip",
+                new HashMap<String, Object>() {{
+                    put(GeoIPPrepperConfig.DATABASE_PATH, TEST_DATABASE_PATH);
+                    put(GeoIPPrepperConfig.DATA_SOURCE, PROVIDER_TYPE);
+                }}
+        );
+        maxMindGeoIpProvider = new MaxMindGeoIpProvider(pluginSetting);
+    }
+
+    @Test
+    public void testIpLookup() {
+        LocationData locationData = new LocationData("United Kingdom", "West Berkshire", "Boxford");
+        Assert.assertTrue(new ReflectionEquals(locationData).matches(maxMindGeoIpProvider.getDataFromIp("2.125.160.216")));
```

Comment:
Suggest implementing an `equals()` method in `LocationData` instead of relying on Reflection here to test. 

Then we'd have: `assertEquals(locationData, provider.get(......));`